core/src/main/java/org/lflang/ast/ASTUtils.java

```diff
@@ -153,7 +153,8 @@ public static List<Reactor> getAllReactors(Resource resource) {
    * Get the main reactor defined in the given resource.
    *
    * @param resource the resource to extract reactors from
-   * @return An iterable over all reactors found in the resource
+   * @return The top-level main reactor, or null if there is no main or federated reactor or the
```

Comment:
This comment isn't right either. The return type is `Optional<Reactor>`, meaning that if a reactor is found, its presence can be checked using the `isPresent()` method and retrieved using `get()`.

Reply:
Feel free to provide a better comment. I'm no so familiar with this pattern.  Otherwise, I'll just revert to the previous incorrect comment.

Reply:
```suggestion
   * @return A top-level main reactor presented as an {@code Optional} that is present if it could be found.
```